client/state/themes/actions.js

```diff
@@ -668,3 +672,33 @@ export function deleteTheme( themeId, siteId ) {
 			} );
 	};
 }
+
+/**
+ * Shows dialog asking user to confirm delete of theme
+ * from jetpack site. Deletes theme if user confirms.
+ *
+ * @param {String} themeId -- Theme to delete
+ * @param {Number} siteId -- Site to delete theme from
+ *
+ * @return {Function} Action thunk
+ */
+export function confirmDelete( themeId, siteId ) {
+	return ( dispatch, getState ) => {
+		const { name: themeName } = getTheme( getState(), siteId, themeId );
+		const siteSlug = getSiteSlug( getState(), siteId );
+		accept(
+			i18n.translate(
+				'Are you sure you want to delete %(themeName)s from %(siteSlug)s?',
+				{ args: { themeName, siteSlug }, context: 'Themes: theme delete confirmation dialog' }
+			),
+			( accepted ) => {
+				accepted && dispatch( deleteTheme( themeId, siteId ) );
+			},
+			i18n.translate(
+				'Delete %(themeName)s',
+				{ args: { themeName }, context: 'Themes: theme delete dialog confirm button' }
+			),
+			i18n.translate( 'Back', { context: 'Theme: theme delete dialog back button' } )
```

Comment:
Alternate string suggestion: 
* `translate( 'Back', { context: 'go back (like the back button in a browser)'} )`&mdash; translations: **35**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/theme-delete-ui)